src/test/java/io/appium/java_client/localserver/ServerBuilderTest.java

```diff
@@ -0,0 +1,129 @@
+package io.appium.java_client.localserver;
+
+import io.appium.java_client.service.local.AppiumDriverLocalService;
+import io.appium.java_client.service.local.AppiumServiceBuilder;
+import io.appium.java_client.service.local.flags.GeneralServerFlag;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.openqa.selenium.Platform;
+
+import java.io.*;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+public class ServerBuilderTest {
+
+    private static File findDefaultNode(){
+        String command;
+        if (Platform.getCurrent().is(Platform.WINDOWS))
+            command =  "npm.cmd -g ls --depth=0";
+        else
+            command = "npm -g ls --depth=0";
+
+        Process p = null;
+        String instancePath;
+        try {
+            p = Runtime.getRuntime().exec(command);
+            p.waitFor();
+
+            BufferedReader reader = new BufferedReader(
+                    new InputStreamReader(p.getInputStream()));
+            instancePath = reader.readLine();
+            reader.close();
+        } catch (IOException e) {
+            throw new RuntimeException("Please install node.js!");
+        }
+        catch (InterruptedException e){
+            throw new RuntimeException(e);
+        }
+
+        File f = new File(instancePath + File.separator + "node_modules" + File.separator +
+                "appium" + File.separator + "bin" + File.separator + "appium.js");
+
+        if (!f.exists()){
+            throw new RuntimeException("Please install appium by npm!");
+        }
+        return f;
+    }
+
+    @Test
+    public void checkAbilityToBuildDefaultService(){
+        AppiumDriverLocalService.buildDefaultService();
+    }
+
+    @Test
+    public void checkAbilityToBuildServiceWithDefinedParametersAndNodeSetInProperties(){
+        try {
+            String definedNode = findDefaultNode().getAbsolutePath();
+            System.setProperty(AppiumServiceBuilder.APPIUM_NODE_PROPERTY, definedNode);
+            AppiumDriverLocalService.buildService(new AppiumServiceBuilder().withIPAddress("127.0.0.1").
+                    usingPort(4000).withArgument(GeneralServerFlag.TEMP_DIRECTORY,
+                    "C:/Windows/Temp"));
```

Comment:
It is passees on Mac OS X. But I'll remove this in order to prevent the confusion.